### PR TITLE
Adds support for pointer/string pointers comparison in LiveQuery

### PR DESCRIPTION
**Pull request by flovilmart**

### CODE REVIEW

File: spec/QueryTools.spec.js [hunk 1]
```diff
@@ -495,4 +495,79 @@ describe('matchesQuery', function() {
     expect(matchesQuery(message, q)).toBe(false);
 
   });
+
+  function pointer(className, objectId) {
+    return {__type: 'Pointer', className, objectId };
```
Comment (acinader) [inline spec/QueryTools.spec.js]: add whitespace between { __

File: src/LiveQuery/QueryTools.js [hunk 2]
```diff
@@ -90,6 +90,28 @@ function queryHash(query) {
 }
 
 /**
+ * contains -- Determines wether a constaint in for form of a list
```
Comment (acinader) [inline src/LiveQuery/QueryTools.js]: "whether" and "constraint" are misspelled.  I haven't read the code yet, but I really can't understand this comment.

File: src/LiveQuery/QueryTools.js [hunk 3]
```diff
@@ -90,6 +90,28 @@ function queryHash(query) {
...
+ * contains a particular object.
+ * This also supports, pointer comparion and strings as lightweight pointers
```
Comment (acinader) [inline src/LiveQuery/QueryTools.js]: "comparison" is misspelled.

File: src/LiveQuery/QueryTools.js [hunk 4]
```diff
@@ -90,6 +90,28 @@ function queryHash(query) {
...
+ */
+function contains(compareTo: Array, object: any): boolean {
+  if (object.__type && object.__type === 'Pointer') {
+    for (const i in compareTo) {
+      const ptr = compareTo[i];
+      if (typeof ptr == 'string' && ptr === object.objectId) {
```
Comment (acinader) [inline src/LiveQuery/QueryTools.js]: ===

File: src/LiveQuery/QueryTools.js [hunk 5]
```diff
@@ -90,6 +90,28 @@ function queryHash(query) {
 }
 
 /**
+ * contains -- Determines wether a constaint in for form of a list
+ * contains a particular object.
+ * This also supports, pointer comparion and strings as lightweight pointers
+ */
+function contains(compareTo: Array, object: any): boolean {
```
Comment (acinader) [inline src/LiveQuery/QueryTools.js]: The names are confusing to me.  particularly compare too.

haystack, needle?

**Comment by flovilmart** [comment]
Thanks for the quick review @acinader !

File: src/LiveQuery/QueryTools.js [hunk 6]
```diff
@@ -90,6 +90,30 @@ function queryHash(query) {
 }
 
 /**
+ * contains -- Determines whether a constraint in the form of a list
```
Comment (acinader) [inline src/LiveQuery/QueryTools.js]: how about:

Determines if an object is contained in a list with special handling for Parse pointers.

Comment (flovilmart) [reply src/LiveQuery/QueryTools.js]: LOVE it!

**Comment by flovilmart** [comment]
seems that eslint don't catch issue when doing `typeof undefinedVariable`

File: src/LiveQuery/QueryTools.js [hunk 7]
```diff
@@ -99,8 +99,7 @@ function contains(haystack: Array, needle: any): boolean {
       if (typeof ptr === 'string' && ptr === needle.objectId) {
         return true;
       }
-      if (typeof object !== 'undefined' &&
-          ptr.className === needle.className &&
+      if (ptr.className === needle.className &&
```
Comment (acinader) [inline src/LiveQuery/QueryTools.js]: that seems good, you've already validated that needle is not undefined in the conatining if.

Comment (acinader) [reply src/LiveQuery/QueryTools.js]: not really sure if we should be forcing 

`typeof needle !== 'undefined'`

over 

`if(needle)`

I use the second form in my code.  Obviously could be an issue in the case of boolean but i just ignore that...

Comment (flovilmart) [reply src/LiveQuery/QueryTools.js]: > Obviously could be an issue in the case of boolean but i just ignore that...

if it's a boolean, we'll go to the 'indexOf' case which is correct :)

Comment (acinader) [reply src/LiveQuery/QueryTools.js]: yup, i was musing out loud more generally about if we should enforce a more formal test than just `if(foo)` without advocating that we switch to it.... :).  so safe to ignore me.

Comment (flovilmart) [reply src/LiveQuery/QueryTools.js]: all those bottom values in JS makes it difficult to make it consistent :/

**Pull request merged**